models/helper.go

```diff
@@ -8,13 +8,18 @@ import (
 	"gorm.io/gorm"
 )
 
+// Helper function that adds extra filter clauses to the query, joining on the right set of tables, excluding data where necessary, etc.
+// * It was added here to make it reusable, given these filters need to be added to many of the queries that take in
+//   a list of filters in the form of concept ids and cohort pairs. The one assumption it makes is that the given `query` object already contains
+//   a basic query on a table or view that have been named or aliased as "observation" (see comments in code).
+//   TODO - find a way to check this assumption here. If we have a good way to get the raw SQL from given `query`, we could check with `\bas observation\b`...
```

Comment:
I tried adding a check using https://gorm.io/docs/sql_builder.html#ToSQL, but this method was not found. Maybe I'm on a different version...

Reply:
https://github.com/go-gorm/gorm/commit/8de266b4a7391145e962918abb3a9705c13fd2c8

Reply:
thanks. So [v1.22.3](https://github.com/go-gorm/gorm/releases/tag/v1.22.3) is the oldest. We are currently at [v1.22.0][https://github.com/uc-cdis/cohort-middleware/blob/master/go.mod#L10]...so that explains it. Let me update...

Reply:
looking at the gorm code made me realize I can actually do this in a simpler way by reading `.Table` name from the db query object. I've added this  now.